bestPractice/Test_BestPracticeTriggerHandler.cls

```diff
@@ -0,0 +1,193 @@
+        }
+        insert accountList;
+        
+        // Contact
+        List<Contact> contactList = new List<Contact>();
+        for (Contact contatcInstance : contactListData) {
+            contatcInstance.Email='abc@gmail.com';
+            contactList.add(contatcInstance);
+        }
+        
+        // Checking the governor limits
+        Test.startTest();
+        insert contactList;
+        Test.stopTest();
+        
+        List<Account> accountData = [
+            SELECT
+                Id 
+            FROM 
+                Account
+        ];
+         
+        List<Contact> contactData = [
+            SELECT 
+                AccountId
+            FROM 
+                Contact 
+        ]; 
+        
+        // Check the final result  ??????????????????????       ERROR
+        for (i=0;i<200;i++) {
+            System.assertEquals(accountData[i].Id, contactData[i].AccountId);
+        }    
+    }  
+    
+/////////////////////////////////////////////////////////////////////////////////////////////
+    
+    // Check with  blank account data  
+    static testMethod void contactEmailDomainBlankCheck() {
+        List<Account> accountListData = BestPracticeDataFactory.accountData();
+        List<Contact> contactListData = BestPracticeDataFactory.contactData();
+        
+        // Acccount
+        List<Account> accountList = new List<Account>();
+        for (Account accountInstance : accountListData) {
+            accountInstance.Website='www.gmail.com';
+            accountList.add(accountInstance);
+        }
+        insert accountList;
+        
+        // Checking the governor limits
+        Test.startTest();
+        insert contactListData;
+        Test.stopTest();
+        
+        /*
+        List<Account> accountData = [
+            SELECT
+                Id 
+            FROM 
+                Account
+        ];
+        */ 
+         
+        List<Contact> contactData = [
+            SELECT 
+                AccountId
+            FROM 
+                Contact 
+        ]; 
+        
+        // Check the final result
+        for (i=0;i<200;i++) {
+            System.assert(contactData[i].AccountId == NULL);
+        }      
+    }
+    
+ //////////////////////////////////////////////////////////////////////////////////////////////
+    
+    // Check with  blank contact data
+    static testMethod void accountDomainBlankCheck() {
+        List<Account> accountListData = BestPracticeDataFactory.accountData();
+        List<Contact> contactListData = BestPracticeDataFactory.contactData();
+        
+        insert accountListData;
+        
+        // Contact
+        List<Contact> contactList = new List<Contact>();
+        for (Contact contatcInstance : contactListData) {
+            contatcInstance.Email='abc@gmail.com';
+            contactList.add(contatcInstance);
+        }
+        
+        
+        // Checking the governor limits
```

Comment:
Here we are inserting contact and not checking governor limits. Comment should be appropriate of what we are doing. Please change in all test methods